### DISCUSSION TITLE
UKMO's Improver project

### DISCUSSION THREAD

**Discussion by phillipseamore** [open-meteo/open-meteo#242] [General]
Stumbled upon this and thought it might be of interest. They are doing advanced lapse rates, downscaling wind and plenty other interesting things.

https://github.com/metoppv/improver

**Comment by patrick-zippenfenig** [comment]
Interesting, thanks for sharing! A better wind downscaling for model blending would be nice... however it requires a lot of recalculated fields..

**Comment by phillipseamore** [comment]
It also seems focused on ensembles so you might get some ideas from that as well. There are also some novel interpolation functions, for instance they have solar aware temporal interpolation.

**Reply by patrick-zippenfenig** [reply]
Which variable are they interpolating? temperature? I was thinking about doing the same for 6 hourly temperature data. The issue is that there is a certain delay in temperature and solar radiation...

To interpolate shortwave radiation, I already use a solar aware temporal interpolation, otherwise the solar radiation would be off entirely... See here: https://github.com/open-meteo/open-meteo/blob/main/Sources/App/Helper/Array2DInterpolation.swift#L156

**Reply by phillipseamore** [reply]
They are scaling by the elevation of the sun. https://github.com/metoppv/improver/blob/a876a94a5f5e3f1454802df0f4f23bf173d34b19/improver/utilities/temporal_interpolation.py#L262

Except for a few tests there aren't really many examples or indications of how the project is used in production.

**Reply by patrick-zippenfenig** [reply]
At first sight, they do not consider that data is backwards averaged. This approach is fine with instantaneous values, but in most weather models solar radiation is backwards averaged. To do it correctly, either supersampling can be used (computationally very expensive) or the sun angle equations need to be solved as an integral. Technically you cannot solve the integral directly, but I figured out an approximation which is correct while the sun is above horizon: https://github.com/open-meteo/open-meteo/blob/ab82fa388ff8036306ff73d606ffcfcae2982148/Sources/App/Helper/Zensun.swift#L126

If their code is actually used with supersampling to ~5 minutes instead of hourly data, then it is probably ok.
